lib/iris/coords.py

```diff
@@ -516,13 +516,26 @@ def copy(self, points=None, bounds=None):
         .. note:: If the points argument is specified and bounds are not, the
                   resulting coordinate will have no bounds.
 
+        .. deprecated:: 1.12
```

Comment:
On a completely seperate issue from deprecations etc...
I'm a bit bothered about the scope of the new behaviour..

Firstly, I'm not convinced that it is expected for a method called "copy" to return a view at all.
I'm thinking that it would make more sense to have ```.copy()``` return a full copy, and use `[:]` for a toplevel-copy.
Isn't that what numpy does, so that is what we should emulate ?
E.G.
```
>>> a=np.arange(10)
>>> b=a[2:4]
>>> b[:] = 0
>>> a
array([0, 1, 0, 0, 4, 5, 6, 7, 8, 9])
>>> b=a.copy()
>>> b[:] = -1
>>> a
array([0, 1, 0, 0, 4, 5, 6, 7, 8, 9])
>>> 
```

Secondly, as I've mentioned it shouldn't ```Coord.__getitem__``` be returning copies ??
(that is, ***instead***, if you accept the previous...)